index.js

```diff
@@ -23,17 +65,28 @@ module.exports = {
       return "<!-- EMBER_CLI_FASTBOOT_TITLE -->";
     }
 
-    if (type === 'vendor-prefix') {
-      return '// Added from ember-cli-fastboot \n' +
-             'EmberENV.FEATURES = EmberENV.FEATURES || {};\n' +
-             'EmberENV.FEATURES["ember-application-visit"] = true;\n';
+    if (type === 'app-boot') {
+      return fastbootAppModule(config.modulePrefix);
     }
   },
 
-  included: function() {
+  treeForApp: function(tree) {
     if (process.env.EMBER_CLI_FASTBOOT) {
-      this.app.options.storeConfigInMeta = false;
-      process.env.EMBER_CLI_FASTBOOT_APP_NAME = this.app.name;
+      return new Funnel(tree, {
+        annotation: 'Funnel: Remove browser-only initializers',
+        exclude: [
+          'initializers/browser/*',
+          'instance-initializers/browser/*'
+        ]
+      });
+    } else {
+      return new Funnel(tree, {
+        annotation: 'Funnel: Remove server-only initializers',
+        exclude: [
+          'initializers/server/*',
+          'instance-initializers/server/*'
+        ]
+      });
     }
```

Comment:
@rwjblue @stefanpenner how does that look? with this I could complete remove the `FastBoot` global we inject into the sandbox.

By the way, if this works out for us, we might want to move it into the `preprocessTree` hook so that apps and other addons can use the same pattern